hyperspy/learn/mva.py

```diff
@@ -1317,22 +1369,679 @@ def _estimate_elbow_position(self, curve_values):
+            clus_index = np.where(labels == i)
+            clustersizes[i] = labels[np.where(labels == i)].shape[0]
+            # if using the pca components
+            if use_decomposition_for_centers:
+                # pca clustered...
+                a = self.learning_results.loadings[clus_index][:,
+                                                       0:number_pca_components]
+                b = self.learning_results.factors[:, 0:number_pca_components].T
+                center = np.dot(a, b).sum(axis=0)
+                cluster_centers[i, :] = cluster_centers[i, :] + center
+                cluster_centers[i, :] = cluster_centers[i, :] 
+            else:
+                cluster_centers[i, :] = clusterdata[clus_index].sum(axis=0)
+
+        # this sorts the labels based on clustersize for high to low
+        # i.e. point with laster number of points first
+        idx = np.argsort(clustersizes)[::-1]
+        lut = np.zeros_like(idx)
+        lut[idx] = np.arange(n_clusters)
+        sorted_labels = lut[labels]
+        shape = (n_clusters, self.data.shape[0])
+        cluster_labels = np.full(shape, np.nan)
+        sorted_cluster_centers = np.zeros_like(cluster_centers)            
+        # now create the labels from these sorted labels
+        for i in range(n_clusters):
+            cluster_labels[i, navigation_mask] = \
+                np.where(sorted_labels == i, 1, 0)
+            sorted_cluster_centers[i,:] = cluster_centers[lut[i],:]\
+                /clustersizes[lut[i]]                
+
+
+
+        return sorted_labels,cluster_labels, sorted_cluster_centers
+
+    def evaluate_number_of_clusters(self,
+                                    max_clusters=12,
+                                    scaling="minmax",
+                                    scaling_kwargs={},
+                                    use_decomposition_results=True,
+                                    use_decomposition_for_centers=False,
+                                    number_pca_components=None,
+                                    navigation_mask=None,
+                                    signal_mask=None,
+                                    algorithm='kmeans',
+                                    metric="gap",
+                                    n_ref=10,
+                                    **kwargs):
+        """
+        Performs cluster analysis of a signal for cluster sizes ranging from
+        n_clusters =2 to max_clusters ( default 15)
+
+        For each cluster it evaluates the silhouette score which is a metric of
+        how well seperated the clusters are. Maximima or peaks in the scores
+        indicate good choices for cluster sizes.
+
+
+        Parameters
+        ----------
+        max_clusters : int, default 15
+            Max number of clusters to use. The method will scan from 2 to
+            max_clusters
+        scaling : {"standard","norm","minmax" or scikit learn scaling method}
```

Comment:
Again, supporting "None" would be good - also the default value of "minmax" is inconsistent to use of "standard" above

Reply:
None can be passed - just wasn't documented - Added documentation and tweaked None check in scaling method